Engine/Module/ModuleResourceManager.cpp

```diff
@@ -291,6 +349,48 @@ uint32_t ModuleResourceManager::CreateFromData(FileData data, const std::string&
 	return InternalImport(*created_asset_file_path);
 }
 
+void ModuleResourceManager::LoaderThread()
+{
+	while(loading_threads_active)
+	{
+		if(!loading_textures_queue.Empty())
+		{
+			TextureLoadJob load_job;
+			if(loading_textures_queue.TryPop(load_job))
+			{
+				//Check if resource is already on cache
+				if(!load_job.already_in_cache)
+				{
+					ResourceManagement::LoadThread<Texture>(load_job.uuid, load_job.exported_file_data, load_job.loaded_data);
```

Comment:
Here instead of the LoadThread call if you have all the gameobjects and components already on the scene, and you store pairs of components pointer and resources uuid, you can do a call like component->LoadResource(uuid); maybe you can add also the type here and do a switch inside if needed for componentes that have more than one kind of resources or add a vector like component->LoadResource(vector<uuuid, type>);